src/libraries/Common/src/System/TimeProvider.cs

```diff
@@ -174,12 +201,23 @@ public override ITimer CreateTimer(TimerCallback callback, object? state, TimeSp
             /// </remarks>
             private sealed class SystemTimeProviderTimer : ITimer
```

Comment:
SystemTimeProviderTimer exists to shield TimerQueueTimer from being returned directly.  That doesn't apply downlevel where we're just using Timer directly. Thus SystemTimeProviderTimer can be left alone and only used at all in .NET 8+, and then downlevel CreateTimer would create and return a Timer rather than a SystemTimeProviderTimer.

Reply:
Ah, ignore that, we still need to wrap it because of the ITimer interface.